File: app/src/main/java/io/eodc/planit/activity/MainActivity.java
package io.eodc.planit.activity;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.support.v4.content.ContextCompat;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.preference.PreferenceManager;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;

import com.aurelhubert.ahbottomnavigation.AHBottomNavigation;
import com.aurelhubert.ahbottomnavigation.AHBottomNavigationAdapter;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import io.eodc.planit.BuildConfig;
import io.eodc.planit.R;
import io.eodc.planit.fragment.AddAssignmentFragment;
import io.eodc.planit.fragment.CalendarFragment;
import io.eodc.planit.fragment.HomeFragment;
import io.eodc.planit.fragment.PlannerFragment;
import timber.log.Timber;

/**
 * Activity that the user mostly interacts with
 *
 * @author 2n
 */
public class MainActivity extends AppCompatActivity {

    @BindView(R.id.bottom_navigation)   AHBottomNavigation      mBottomNav;
    @BindView(R.id.create_fab)          FloatingActionButton    mFab;
    @BindView(R.id.toolbar)             Toolbar                 mToolbar;

    private FragmentManager         mFragmentManager;

    @OnClick(R.id.create_fab) void handleCreateFab() {
        if (getFragmentManager() != null) {
            new AddAssignmentFragment().show(getSupportFragmentManager(), null);
        }
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);

        mFragmentManager = getSupportFragmentManager();

        if (BuildConfig.DEBUG) Timber.plant(new Timber.DebugTree());

        setSupportActionBar(mToolbar);
        mToolbar.inflateMenu(R.menu.main_menu);

        setupBottomNavigation();
        mFab.hide();

        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this);
        String initScreen = sharedPreferences.getString(getString(R.string.pref_init_page_key), getString(R.string.pref_init_page_home_value));

        mBottomNav.setCurrentItem(-1); // Set nav to out of bounds so that callback for home is fired. Makes for consistent code in the if-block
        if (initScreen.equals(getString(R.string.pref_init_page_home_value))) {
            mBottomNav.setCurrentItem(0);
        } else if (initScreen.equals(getString(R.string.pref_init_page_planner_value))) {
            mBottomNav.setCurrentItem(1);
        } else if (initScreen.equals(getString(R.string.pref_init_page_calendar_value))) {
            mBottomNav.setCurrentItem(2);
        }

    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.main_menu, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();

        switch (id) {
            case R.id.mnu_settings:
                Intent intent = new Intent(this, SettingsActivity.class);
                startActivity(intent);
                return true;
        }
        return false;
    }

    private void setupBottomNavigation() {
        AHBottomNavigationAdapter bottomNavAdapter = new AHBottomNavigationAdapter(this, R.menu.bottom_menu);
        mBottomNav.setAccentColor(ContextCompat.getColor(this, R.color.colorAccent));
        bottomNavAdapter.setupWithBottomNavigation(mBottomNav);
        mBottomNav.setOnTabSelectedListener((position, wasSelected) -> {
            if (!wasSelected) {
                FragmentTransaction transaction = mFragmentManager.beginTransaction()
                        .setCustomAnimations(R.anim.fade_in, R.anim.fade_out);
                switch (position) {
                    case 0:
                        mFab.hide();
                        transaction.replace(R.id.content_fragment, new HomeFragment()).commit();
                        break;
                    case 1:
                        mFab.show();
                        transaction.replace(R.id.content_fragment, new PlannerFragment()).commit();
                        break;
                    case 2:
                        mFab.hide();
                        transaction.replace(R.id.content_fragment, new CalendarFragment()).commit();
                        break;
                }
            }
            return true;
        });

    }
}

File: app/src/main/java/io/eodc/planit/activity/MainActivity.kt
package io.eodc.planit.activity

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProviders
import androidx.preference.PreferenceManager
import com.google.android.material.bottomnavigation.BottomNavigationView
import io.eodc.planit.BuildConfig
import io.eodc.planit.R
import io.eodc.planit.db.Subject
import io.eodc.planit.fragment.AddAssignmentFragment
import io.eodc.planit.fragment.CalendarFragment
import io.eodc.planit.fragment.HomeFragment
import io.eodc.planit.fragment.PlannerFragment
import io.eodc.planit.model.SubjectListViewModel
import kotlinx.android.synthetic.main.activity_main.*
import timber.log.Timber

/**
 * Activity that the user mostly interacts with
 *
 * @author 2n
 */
class MainActivity : AppCompatActivity(), BottomNavigationView.OnNavigationItemSelectedListener {
    var subjects: List<Subject>? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        Thread {
            subjects = ViewModelProviders.of(this@MainActivity)
                    .get<SubjectListViewModel>(SubjectListViewModel::class.java)
                    .subjects
        }.start()

        if (BuildConfig.DEBUG) Timber.plant(Timber.DebugTree())

        setSupportActionBar(tb)
        tb.inflateMenu(R.menu.main_menu)

        fabCreateAssignment.hide()
        fabCreateAssignment.setOnClickListener {
            if (supportFragmentManager != null) {
                AddAssignmentFragment().show(supportFragmentManager, null)
            }
        }
        bottomNavigation.setOnNavigationItemSelectedListener(this)

        val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this)
        val initScreen = sharedPreferences.getString(getString(R.string.pref_init_page_key), getString(R.string.pref_init_page_home_value))

        val transaction = supportFragmentManager.beginTransaction()
                .setCustomAnimations(R.anim.fade_in, R.anim.fade_out)

        when (initScreen) {
            getString(R.string.pref_init_page_home_value) -> {
                bottomNavigation.selectedItemId = R.id.bnv_home
                transaction.replace(R.id.fragment_content, HomeFragment()).commit()
            }
            getString(R.string.pref_init_page_planner_value) -> {
                bottomNavigation.selectedItemId = R.id.bnv_planner
                fabCreateAssignment.show()
                transaction.replace(R.id.fragment_content, PlannerFragment()).commit()
            }
            getString(R.string.pref_init_page_calendar_value) -> {
                bottomNavigation.selectedItemId = R.id.bnv_calendar
                transaction.replace(R.id.fragment_content, CalendarFragment()).commit()
            }
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.main_menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val id = item.itemId

        if (id == R.id.mnu_settings) {
            val intent = Intent(this, SettingsActivity::class.java)
            startActivity(intent)
            return true
        }
        return false
    }


    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        val id = item.itemId

        if (id != bottomNavigation.selectedItemId) {
            val transaction = supportFragmentManager.beginTransaction()
                    .setCustomAnimations(R.anim.fade_in, R.anim.fade_out)
            when (id) {
                R.id.bnv_home -> {
                    fabCreateAssignment.hide()
                    transaction.replace(R.id.fragment_content, HomeFragment()).commit()
                }
                R.id.bnv_planner -> {
                    fabCreateAssignment.show()
                    transaction.replace(R.id.fragment_content, PlannerFragment()).commit()
                }
                R.id.bnv_calendar -> {
                    fabCreateAssignment.hide()
                    transaction.replace(R.id.fragment_content, CalendarFragment()).commit()
                }
            }
            return true
        }
        return false
    }
}
